src/models/dataStore.js

```diff
@@ -0,0 +1,69 @@
+import _ from "lodash";
+
+const dataStoreNamespace = "metatada-synchronization";
+const instancesKey = "instances";
+
+async function getDataStore(d2, dataStoreKey, defaultValue = []) {
+    const existsNamespace = await d2.dataStore.has(dataStoreNamespace);
+    if (!existsNamespace) {
+        const dataStore = await d2.dataStore.create(dataStoreNamespace);
+        await dataStore.set(dataStoreKey, defaultValue);
+        return defaultValue;
+    } else {
+        const dataStore = await d2.dataStore.get(dataStoreNamespace);
+        return await dataStore.get(dataStoreKey);
+    }
+}
+
+async function saveDataStore(d2, dataStoreKey, newValue) {
```

Comment:
getDataStore / saveDataStore -> same code except last line, maybe abstract common code to function.